### PR TITLE
Running a word count on a compressed fastq file now runs wc on the un…

### LINKED ISSUE #4144: Report on the number of lines of fastqsanger.gz datatype
When we have a number of lines count on a fastqsanger.gz (i.e on the right column of history executions), the count is from the compressed file. This makes little sense, it would be more helpful if it was from the uncompressed file (e.g. to compare paired reads).

In case this is not clear, imagine a file seqs.fq.gz:
```
$ zless seqs.fq.gz |wc
 989364 1236705 88366482
$ wc seqs.fq.gz 
   88020   515400 24417349
```

Galaxy reports 88020, I think it would make more sense to report 989364.

### PR DESCRIPTION
**Pull request by dpryan79**
…compressed file

This should fix #4144.

### CODE REVIEW

**Comment by martenson** [comment]
Do I understand it correctly that the issue here is that `wc` can take 'any' datatype thus won't take advantage of the implicit conversion? If so are there other tools that behave like this?

**Comment by dpryan79** [comment]
Yes, my understanding of #4144 is that it'd be useful to have `wc` properly handle compressed input. Regarding other data types, the fastq ones are the only ones I know of that stay compressed in Galaxy and don't already have dedicated tools to handle things like this (BAM would be an example of a compressed format for which there are already dedicated tools).

File: tools/filters/wc_gnu.xml [hunk 1]
```diff
@@ -12,11 +12,20 @@
             echo "#${ "\t".join( [ i for i in $arg_order if i in $args ] ) }" &gt; $out_file1
             &amp;&amp;
         #end if
+
+        ##Handle compressed (fastq) input
+        #if $input1.is_of_type("fastq.gz"):
+            zcat $input1 |
```
Comment (nsoranzo) [inline tools/filters/wc_gnu.xml]: Single-quote `$input1` here and below?

Comment (dpryan79) [reply tools/filters/wc_gnu.xml]: Yes, thanks for catching this!

**Comment by nsoranzo** [comment]
Thanks @dpryan79!

**Comment by bgruening** [comment]
Nice! :)

**Comment by martenson** [comment]
@dpryan79 my question was whether there are more wc-like tools that will take fastq.gz and do unreasonable things instead of decompressing it

**Comment by dpryan79** [comment]
@martenson Ah, probably? `wc` is the only one I know of that people are likely to use though.

**Pull request merged**